AllergiesTest.cs giving error when run.

I'm trying to do the C# exercism exercise Allergies.
When I run the usual `dotnet test`, I get the following error:
```
error CS0111: Type 'AllergiesTest' already defines a member called 'Not_allergic_to_anything' with the same parameter types 
```
There is one of these errors for every test in the AllergiesTest.cs file.
So far this is the only exercise that has given this error.

Edit: I figured out the problem.  There are several tests that all share the name 'Not_allergic_to_anything', as well as 'Allergic_to_everything'.  The test suite also expects a method called `AllergicTo`, while the provided template names this method `IsAllergicTo`.  When I changed the method names the test suite ran without issues.

--- a/.gitignore
+++ b/.gitignore
@@ -255,6 +255,7 @@ packages/
 
 .vscode/
 .idea/
+.ionide/
 
 *.un~
 

--- a/exercises/allergies/AllergiesTest.cs
+++ b/exercises/allergies/AllergiesTest.cs
@@ -5,283 +5,283 @@ using Xunit;
 public class AllergiesTest
 {
     [Fact]
-    public void Not_allergic_to_anything()
+    public void Testing_for_eggs_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Eggs));
+        Assert.False(sut.IsAllergicTo(Allergen.Eggs));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_eggs()
+    public void Testing_for_eggs_allergy_allergic_only_to_eggs()
     {
         var sut = new Allergies(1);
-        Assert.True(sut.AllergicTo(Allergen.Eggs));
+        Assert.True(sut.IsAllergicTo(Allergen.Eggs));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_eggs_and_something_else()
+    public void Testing_for_eggs_allergy_allergic_to_eggs_and_something_else()
     {
         var sut = new Allergies(3);
-        Assert.True(sut.AllergicTo(Allergen.Eggs));
+        Assert.True(sut.IsAllergicTo(Allergen.Eggs));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_eggs()
+    public void Testing_for_eggs_allergy_allergic_to_something_but_not_eggs()
     {
         var sut = new Allergies(2);
-        Assert.False(sut.AllergicTo(Allergen.Eggs));
+        Assert.False(sut.IsAllergicTo(Allergen.Eggs));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_eggs_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Eggs));
+        Assert.True(sut.IsAllergicTo(Allergen.Eggs));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_peanuts_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Peanuts));
+        Assert.False(sut.IsAllergicTo(Allergen.Peanuts));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_peanuts()
+    public void Testing_for_peanuts_allergy_allergic_only_to_peanuts()
     {
         var sut = new Allergies(2);
-        Assert.True(sut.AllergicTo(Allergen.Peanuts));
+        Assert.True(sut.IsAllergicTo(Allergen.Peanuts));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_peanuts_and_something_else()
+    public void Testing_for_peanuts_allergy_allergic_to_peanuts_and_something_else()
     {
         var sut = new Allergies(7);
-        Assert.True(sut.AllergicTo(Allergen.Peanuts));
+        Assert.True(sut.IsAllergicTo(Allergen.Peanuts));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_peanuts()
+    public void Testing_for_peanuts_allergy_allergic_to_something_but_not_peanuts()
     {
         var sut = new Allergies(5);
-        Assert.False(sut.AllergicTo(Allergen.Peanuts));
+        Assert.False(sut.IsAllergicTo(Allergen.Peanuts));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_peanuts_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Peanuts));
+        Assert.True(sut.IsAllergicTo(Allergen.Peanuts));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_shellfish_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Shellfish));
+        Assert.False(sut.IsAllergicTo(Allergen.Shellfish));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_shellfish()
+    public void Testing_for_shellfish_allergy_allergic_only_to_shellfish()
     {
         var sut = new Allergies(4);
-        Assert.True(sut.AllergicTo(Allergen.Shellfish));
+        Assert.True(sut.IsAllergicTo(Allergen.Shellfish));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_shellfish_and_something_else()
+    public void Testing_for_shellfish_allergy_allergic_to_shellfish_and_something_else()
     {
         var sut = new Allergies(14);
-        Assert.True(sut.AllergicTo(Allergen.Shellfish));
+        Assert.True(sut.IsAllergicTo(Allergen.Shellfish));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_shellfish()
+    public void Testing_for_shellfish_allergy_allergic_to_something_but_not_shellfish()
     {
         var sut = new Allergies(10);
-        Assert.False(sut.AllergicTo(Allergen.Shellfish));
+        Assert.False(sut.IsAllergicTo(Allergen.Shellfish));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_shellfish_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Shellfish));
+        Assert.True(sut.IsAllergicTo(Allergen.Shellfish));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_strawberries_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Strawberries));
+        Assert.False(sut.IsAllergicTo(Allergen.Strawberries));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_strawberries()
+    public void Testing_for_strawberries_allergy_allergic_only_to_strawberries()
     {
         var sut = new Allergies(8);
-        Assert.True(sut.AllergicTo(Allergen.Strawberries));
+        Assert.True(sut.IsAllergicTo(Allergen.Strawberries));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_strawberries_and_something_else()
+    public void Testing_for_strawberries_allergy_allergic_to_strawberries_and_something_else()
     {
         var sut = new Allergies(28);
-        Assert.True(sut.AllergicTo(Allergen.Strawberries));
+        Assert.True(sut.IsAllergicTo(Allergen.Strawberries));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_strawberries()
+    public void Testing_for_strawberries_allergy_allergic_to_something_but_not_strawberries()
     {
         var sut = new Allergies(20);
-        Assert.False(sut.AllergicTo(Allergen.Strawberries));
+        Assert.False(sut.IsAllergicTo(Allergen.Strawberries));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_strawberries_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Strawberries));
+        Assert.True(sut.IsAllergicTo(Allergen.Strawberries));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_tomatoes_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Tomatoes));
+        Assert.False(sut.IsAllergicTo(Allergen.Tomatoes));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_tomatoes()
+    public void Testing_for_tomatoes_allergy_allergic_only_to_tomatoes()
     {
         var sut = new Allergies(16);
-        Assert.True(sut.AllergicTo(Allergen.Tomatoes));
+        Assert.True(sut.IsAllergicTo(Allergen.Tomatoes));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_tomatoes_and_something_else()
+    public void Testing_for_tomatoes_allergy_allergic_to_tomatoes_and_something_else()
     {
         var sut = new Allergies(56);
-        Assert.True(sut.AllergicTo(Allergen.Tomatoes));
+        Assert.True(sut.IsAllergicTo(Allergen.Tomatoes));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_tomatoes()
+    public void Testing_for_tomatoes_allergy_allergic_to_something_but_not_tomatoes()
     {
         var sut = new Allergies(40);
-        Assert.False(sut.AllergicTo(Allergen.Tomatoes));
+        Assert.False(sut.IsAllergicTo(Allergen.Tomatoes));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_tomatoes_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Tomatoes));
+        Assert.True(sut.IsAllergicTo(Allergen.Tomatoes));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_chocolate_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Chocolate));
+        Assert.False(sut.IsAllergicTo(Allergen.Chocolate));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_chocolate()
+    public void Testing_for_chocolate_allergy_allergic_only_to_chocolate()
     {
         var sut = new Allergies(32);
-        Assert.True(sut.AllergicTo(Allergen.Chocolate));
+        Assert.True(sut.IsAllergicTo(Allergen.Chocolate));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_chocolate_and_something_else()
+    public void Testing_for_chocolate_allergy_allergic_to_chocolate_and_something_else()
     {
         var sut = new Allergies(112);
-        Assert.True(sut.AllergicTo(Allergen.Chocolate));
+        Assert.True(sut.IsAllergicTo(Allergen.Chocolate));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_chocolate()
+    public void Testing_for_chocolate_allergy_allergic_to_something_but_not_chocolate()
     {
         var sut = new Allergies(80);
-        Assert.False(sut.AllergicTo(Allergen.Chocolate));
+        Assert.False(sut.IsAllergicTo(Allergen.Chocolate));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_chocolate_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Chocolate));
+        Assert.True(sut.IsAllergicTo(Allergen.Chocolate));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_pollen_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Pollen));
+        Assert.False(sut.IsAllergicTo(Allergen.Pollen));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_pollen()
+    public void Testing_for_pollen_allergy_allergic_only_to_pollen()
     {
         var sut = new Allergies(64);
-        Assert.True(sut.AllergicTo(Allergen.Pollen));
+        Assert.True(sut.IsAllergicTo(Allergen.Pollen));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_pollen_and_something_else()
+    public void Testing_for_pollen_allergy_allergic_to_pollen_and_something_else()
     {
         var sut = new Allergies(224);
-        Assert.True(sut.AllergicTo(Allergen.Pollen));
+        Assert.True(sut.IsAllergicTo(Allergen.Pollen));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_pollen()
+    public void Testing_for_pollen_allergy_allergic_to_something_but_not_pollen()
     {
         var sut = new Allergies(160);
-        Assert.False(sut.AllergicTo(Allergen.Pollen));
+        Assert.False(sut.IsAllergicTo(Allergen.Pollen));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_pollen_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Pollen));
+        Assert.True(sut.IsAllergicTo(Allergen.Pollen));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Not_allergic_to_anything()
+    public void Testing_for_cats_allergy_not_allergic_to_anything()
     {
         var sut = new Allergies(0);
-        Assert.False(sut.AllergicTo(Allergen.Cats));
+        Assert.False(sut.IsAllergicTo(Allergen.Cats));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_only_to_cats()
+    public void Testing_for_cats_allergy_allergic_only_to_cats()
     {
         var sut = new Allergies(128);
-        Assert.True(sut.AllergicTo(Allergen.Cats));
+        Assert.True(sut.IsAllergicTo(Allergen.Cats));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_cats_and_something_else()
+    public void Testing_for_cats_allergy_allergic_to_cats_and_something_else()
     {
         var sut = new Allergies(192);
-        Assert.True(sut.AllergicTo(Allergen.Cats));
+        Assert.True(sut.IsAllergicTo(Allergen.Cats));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_something_but_not_cats()
+    public void Testing_for_cats_allergy_allergic_to_something_but_not_cats()
     {
         var sut = new Allergies(64);
-        Assert.False(sut.AllergicTo(Allergen.Cats));
+        Assert.False(sut.IsAllergicTo(Allergen.Cats));
     }
 
     [Fact(Skip = "Remove to run test")]
-    public void Allergic_to_everything()
+    public void Testing_for_cats_allergy_allergic_to_everything()
     {
         var sut = new Allergies(255);
-        Assert.True(sut.AllergicTo(Allergen.Cats));
+        Assert.True(sut.IsAllergicTo(Allergen.Cats));
     }
 
     [Fact(Skip = "Remove to run test")]

--- a/generators/Exercises/Generators/Allergies.cs
+++ b/generators/Exercises/Generators/Allergies.cs
@@ -11,9 +11,11 @@ namespace Exercism.CSharp.Exercises.Generators
         {
             testMethod.ConstructorInputParameters = new[] { "score" };
             testMethod.TestedMethodType = TestedMethodType.InstanceMethod;
-            
+
             if (testMethod.Property == "allergicTo")
             {
+                testMethod.TestMethodName = testMethod.TestMethodNameWithPath;
+                testMethod.TestedMethod = "IsAllergicTo";
                 testMethod.Input["item"] = RenderAllergen(testMethod.Input["item"]);
             }
             else

